src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala

```diff
@@ -455,6 +453,17 @@ final case class ProcessApiServiceImpl(
     }
   }
 
+  def getGetEserviceAndSelfcareId(
+    eservice: client.model.EService
+  )(implicit contexts: Seq[(String, String)]): Future[(client.model.EService, UUID)] = eservice
+    .pure[Future]
+    .mproduct(eservice =>
+      tenantManagementService
+        .getTenant(eservice.producerId)
+        .flatMap(_.selfcareId.toFuture(MissingSelfcareId))
+        .flatMap(_.toFutureUUID)
```

Comment:
this is a `String` for us, we should not convert it to `UUID`.
If you need this on line 435, I suggest something more like
```
organizationsDetails.found.find(_.id.toString == selfcareId)
```